src/storage/datasets_index.rs

```diff
@@ -43,6 +44,60 @@ impl DatasetsIndex {
                 .collect()
         })
     }
+    pub fn from(assigned_data: Vec<crate::util::assignment::Dataset>, headers: HashMap<String, String>) -> Self {
```

Comment:
Does it store all the existing chunks now? It will probably take gigabytes of memory. Can we only keep the chunks assigned to this worker?

Reply:
Discussed and postponed.